hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -2332,37 +2332,76 @@ private boolean shouldForbidMajorCompaction() {
   }
 
   /**
-   * We are trying to remove / relax the region read lock for compaction. Let's see what are the
```

Comment:
This comment was helpful for understanding the compaction logic. I cleaned up the formatting to how it was before spotless, using proper javadoc syntax.